### ISSUE TITLE
Apply `record-brace-space` to datatype declarations

### ISSUE THREAD

**Issue by LightAndLight** [fourmolu/fourmolu#442]
**Is your feature request specific to Fourmolu?**
- [x] Yes, this feature is not relevant to Ormolu

**Describe the current + desired formatting**

`record-brace-space` only applies to single-line record expressions. I propose that it also apply to single-line record constructor definitions.

Examples:

<table>
<thead>
<th>Input</th>
<th>

`record-brace-space`

</th>
<th>Current output</th>
<th>Proposed output</th>
</thead>
<tbody>

<tr>
<td>

```haskell
x=X{a=b,c=d}
```

</td>
<td>

`false`

</td>
<td>

```haskell
x = X{ a = b, c = d }
```

</td>
<td>

*unchanged*

</td>
</tr>

<tr>
<td>

```haskell
data  X=X{a::A,b::B}
```

</td>
<td>

`false`

</td>
<td>

```haskell
data X = X { a :: A, b :: B }
```

</td>
<td>

```haskell
data X = X{ a :: A, b :: B }
```

</td>
</tr>

<tr>
<td>

```haskell
x=X{a=b,c=d}
```

</td>
<td>

`true`

</td>
<td>

```haskell
x = X { a = b, c = d }
```

</td>
<td>

*unchanged*

</td>
</tr>

<tr>
<td>

```haskell
data  X=X{a::A,b::B}
```

</td>
<td>

`true`

</td>
<td>

```haskell
data X = X { a :: A, b :: B }
```

</td>
<td>

*unchanged*

</td>
</tr>
</tbody>
</table>

If you're willing to include this change then I'll work on it.

**Comment by brandonchinn178** [comment]
I'm surprised someone's actually using `record-brace-space: false` 😅 AFAIK that option was really only for compatibility with Ormolu.

I think my main concern is that this would break Ormolu compatibility, since there'd be no way to recover vanilla Ormolu behavior. This would have to be a new option, or upstreamed in Ormolu. Maybe see what Ormolu thinks?

**Comment by LightAndLight** [comment]
I use `record-brace-space: false` because spaces are generally used for function applications. I see `X{...}` as a single syntactic element because `{...}` can't exist on its own.

> Maybe see what Ormolu thinks?

:+1:

**Comment by brandonchinn178** [comment]
Oh wait sorry, I messed up the option. true is ormolu behavior, false is the default + our behavior.

Yes, I'd be in favor of just making this change. I'd be interested to see if this change breaks the cabal compat check in CI. If it doesn't, I think it's a rare enough case to warrant a breaking change. @georgefst thoughts?

**Comment by georgefst** [comment]
Yeah, I suppose the Cabal check is the closest thing we have to an analysis of how much code will change in the wild. And this would hardly be the first time we've broken _some_ code, as much as I'd like to do it less and less over time. Plus, the existing behaviour is clearly a mistake (by me I guess - I never really define record types on a single line, which might be why I didn't think of it).

**Comment by georgefst** [comment]
> Plus, the existing behaviour is clearly a mistake

Actually, I did explicitly choose to do it this way (https://github.com/fourmolu/fourmolu/pull/26#issuecomment-695823796), but not for any particularly good reason. And that was back in the very early days, when there weren't really any co-maintainers to argue with me, and barely even any users.
